### PR TITLE
Subscription

### PR DESCRIPTION
**Pull request by CanerYesildag**
Subscription API endpoints added for java client

### CODE REVIEW

File: src/main/java/com/iyzipay/model/subscription/enumtype/SubscriptionInterval.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package com.iyzipay.model.subscription.enumtype;
+
+public enum SubscriptionInterval {
```
Comment (erhmutlu) [inline src/main/java/com/iyzipay/model/subscription/enumtype/SubscriptionInterval.java]: SubscriptionPaymentInterval

File: src/main/java/com/iyzipay/model/subscription/enumtype/SubscriptionProductType.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package com.iyzipay.model.subscription.enumtype;
+
+public enum SubscriptionProductType {
```
Comment (erhmutlu) [inline src/main/java/com/iyzipay/model/subscription/enumtype/SubscriptionProductType.java]: buna gerek yok

File: src/test/java/com/iyzipay/sample/subscription/SubscriptionCardUpdateCheckoutFormInitializeSample.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class SubscriptionCardUpdateCheckoutFormInitializeSample extends Sample {
+
+    @Test
+    public void should_initialize_subscription_card_update_checkout_form() {
+        InitializeSubscriptionCardUpdateCheckoutFormRequest cardUpdateCheckoutFormRequest = new InitializeSubscriptionCardUpdateCheckoutFormRequest();
+        cardUpdateCheckoutFormRequest.setCallbackUrl("www.google.com");
```
Comment (erhmutlu) [inline src/test/java/com/iyzipay/sample/subscription/SubscriptionCardUpdateCheckoutFormInitializeSample.java]: https://www.merchant.com/callback

File: src/main/java/com/iyzipay/model/subscription/SubscriptionPricingPlan.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+    public static SubscriptionPricingPlan update(String subscriptionPricingPlanReferenceCode, UpdateSubscriptionPricingPlanRequest request, Options options) {
+        String uri = options.getBaseUrl() + "/v2/subscription/pricing-plans/" + subscriptionPricingPlanReferenceCode;
+        return HttpClient.create().post(uri,
+                getHttpHeadersV2(uri, request, options),
+                request,
+                SubscriptionPricingPlan.class);
+    }
+
+    public static IyzipayResource delete(String subscriptionPricingPlanReferenceCode, Options options) {
```
Comment (ferhat-taslicukur) [inline src/main/java/com/iyzipay/model/subscription/SubscriptionPricingPlan.java]: Burası IyzipayResource olmayacak sanırım

**Pull request merged**